### DISCUSSION TITLE
ViewModel WhenActivated is not called (resp. Compiler error CS0121 "Ambiguous call")

### DISCUSSION THREAD

**Discussion by adaxer** [reactiveui/ReactiveUI#3504] [Q&A] [answered]
Hi, I am evaluating ReactiveUI and created a vanilla app with only one ViewModel and View (net6.0-windows, ReactiveUI v18.4.26)
- In App() I call the Locator.CurrentMutable.RegisterViewsForViewModels(Assembly.GetCallingAssembly());  
- The ViewModel (MainViewModel) implements IActivatableViewModel 
- The View implements IViewFor<MainViewModel>.
According to the ReactiveUI documentation, when the View is activated, the WhenActivated method of the ViewModel should be called.

Problem1: The compiler complains about an ambiguous call in the ViewModel:
(...) Error	CS0121	The call is ambiguous between the following methods or properties: 'ViewForMixins.WhenActivated(IActivatableViewModel, Action _of_ Action _of_ IDisposable)' and 'ViewForMixins.WhenActivated(IActivatableViewModel, Action _of_ CompositeDisposable)'	(...)
(Sorry for the VB resemblance, somehow the editor crippled the parameters)

Problem2: If I avoid the error by qualifying the args of the call, then none of the Methods is called.

The problem occurs with the current VS2022 version (17.5.2) and with Rider as well
What am I doing wrong? Any pointers would be very much welcome

Attached a zip of the solution

[RxUIIntro.zip](https://github.com/reactiveui/ReactiveUI/files/11043945/RxUIIntro.zip)

**Comment by ChrisPulman** [comment, marked as answer]
Hi @adaxer,
Firstly you need to target windows 10.0.17763.0 + to use the latest ReactiveUI due to compatibility with other dependencies, please change to the following minimum version or above.
<TargetFramework>net6.0-windows10.0.17763.0</TargetFramework>

Please update your viewmodel as follows, the way in which you initialised your property will return a new activator every time.
Once you have done this you should be up and running.
```c#
public class MainViewModel : ReactiveObject, IActivatableViewModel
{
    public MainViewModel()
    {
        Activator = new ViewModelActivator();
        this.WhenActivated((CompositeDisposable d) =>
        {
            Console.WriteLine("WhenActivated");
        });
        this.WhenActivated((Action<IDisposable> a) =>
        {
            Console.WriteLine("WhenActivated");
        });
    }
    public ViewModelActivator Activator { get; }
}
```
I hope that this helps you.

**Reply by adaxer** [reply]
Hi Chris, that worked indeed, thanks very much!
